From: Terry Wilson <twilson@digium.com>
Subject: Don't crash on INFO automon request with no channel
 AST-2011-014. When automon was enabled in features.conf, it was possible
 to crash Asterisk by sending an INFO request if no channel had been
 created yet.
Origin: upstream, http://svnview.digium.com/svn/asterisk?view=revision&revision=347531
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-18805
--- a/channels/chan_sip.c
+++ b/channels/chan_sip.c
@@ -18286,11 +18286,18 @@ static void handle_request_info(struct s
 			per device. I don't want incoming callers to record calls in my
 			pbx.
 		*/
-		/* first, get the feature string, if it exists */
+		
 		struct ast_call_feature *feat;
 		int j;
 		struct ast_frame f = { AST_FRAME_DTMF, };
 
+		if (!p->owner) {        /* not a PBX call */
+			transmit_response(p, "481 Call leg/transaction does not exist", req);
+			sip_scheddestroy(p, DEFAULT_TRANS_TIMEOUT);
+			return;
+		}
+
+		/* first, get the feature string, if it exists */
 		ast_rdlock_call_features();
 		feat = ast_find_call_feature("automon");
 		if (!feat || ast_strlen_zero(feat->exten)) {
